File: data/code_pairs/fortran/597.f90
program factorialTest
    implicit none
    call testFactorial()
contains

    function factorial(n) result(res)
        integer, intent(in) :: n
        integer :: res
        integer :: i
        
        if (n == 0) then
            res = 1
            return
        end if

        res = 1
        do i = 1, n
            res = res * i
        end do
    end function factorial

    subroutine testFactorial()
        integer :: expected, computed

        ! Test 1
        expected = 120  ! 5!
        computed = factorial(5)
        call assertEqual(computed, expected, "Test 1")

        ! Test 2
        expected = 1  ! 0!
        computed = factorial(0)
        call assertEqual(computed, expected, "Test 2")

        ! Add more tests as needed
    end subroutine testFactorial

    subroutine assertEqual(computed, expected, message)
        integer, intent(in) :: computed, expected
        character(len=*), intent(in) :: message
        
        if (computed .eq. expected) then
            print *, message, ": Success"
        else
            print *, message, ": Failure. Expected", expected, "but got", computed
        end if
    end subroutine assertEqual

end program factorialTest

File: data/code_pairs/cpp/597.cpp
#include <iostream>
using namespace std;

int factorial(int n) {
    if (n == 0) {
        return 1;
    }

    int result = 1;
    for (int i = 1; i <= n; ++i) {
        result *= i;
    }
    return result;
}

void assertEqual(int computed, int expected, const string& message) {
    if (computed == expected) {
        cout << message << ": Success" << endl;
    } else {
        cout << message << ": Failure. Expected " << expected << " but got " << computed << endl;
    }
}

void testFactorial() {
    // Test 1
    assertEqual(factorial(5), 120, "Test 1");

    // Test 2
    assertEqual(factorial(0), 1, "Test 2");

    // Add more tests as needed
}

int main() {
    testFactorial();
    return 0;
}